dask_jobqueue/core.py

```diff
@@ -203,12 +203,16 @@ def __init__(
         if shebang is None:
             shebang = dask.config.get("jobqueue.%s.shebang" % config_name)
 
-        if cores is None:
+        if (cores is None) and (memory is None):
```

Comment:
You don't need the parentheses: `if cores is None and memory is None`. It may look weird if you are used to language like `C` but when are used to Python removing parentheses feels less noisy.